cdap-security/src/main/java/co/cask/cdap/security/server/BasicAuthenticationHandler.java

```diff
@@ -35,6 +39,9 @@
   @Override
   protected LoginService getHandlerLoginService() {
     String realmFile = handlerProps.get(Constants.Security.BASIC_REALM_FILE);
+    Path realmFilePath = Paths.get(realmFile);
+    Preconditions.checkArgument(Files.exists(realmFilePath), "File does not exist: %s", realmFilePath);
+    Preconditions.checkArgument(Files.isReadable(realmFilePath), "File is not readable: %s", realmFilePath);
```

Comment:
Isn't that you just need to check `isReadable`? If a file doesn't exists, it won't be readable, right?

Reply:
The error message is more useful (more specific) to the user if we do it like this. Otherwise, you don't know if the file didn't exist or if it was just a permissions issue.

Reply:
ok, although it's quite obvious if the user just look at the path, regardless of what the error message is being shown in CDAP, the `ls` would tell.

Reply:
We've encountered scenarios where a file was on NFS, and so we suspected a temporary NFS outage. An error message here that says the file exists but is not readable would give us hints that it is not an NFS outage.